django/forms/widgets.py

```diff
@@ -573,6 +573,19 @@ def render_options(self, selected_choices):
                 output.append(self.render_option(selected_choices, option_value, option_label))
         return '\n'.join(output)
 
+    def use_required_attribute(self, initial):
+        # If first item has a value, rendering required attribute causes HTML validation error.
+        # More info: https://github.com/validator/validator/issues/47#issuecomment-77534952
+        if self.allow_multiple_selected or hasattr(self, 'renderer'):
+            return super(Select, self).use_required_attribute(initial)
+        else:
+            first_item = next(self.choices.__iter__(), None)
+            first_item_is_empty = first_item and \
```

Comment:
In the case of an empty select (`choices = []`), this will still output the `required` attribute, which is still not valid:

```
>>> class TestForm(Form):
...     some_field = forms.ChoiceField(choices=[])
... 
>>> t = TestForm()
>>> print(t['some_field'])
<select id="id_some_field" name="some_field" required>
</select>
```

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Validation test</title>
</head>
<body>
  <select id="id_some_field" name="some_field" required>
  </select>
</body>
</html>
```
Check it here: https://validator.w3.org/nu/#textarea